crates/core/src/factories/core_fetch/test.rs

```diff
@@ -0,0 +1,409 @@
+use std::{sync::Arc, time::Duration};
+
+use bytes::Bytes;
+use kitsune2_api::{fetch::Fetch, id::Id, AgentId, K2Error, OpId, SpaceId};
+use rand::Rng;
+use tokio::sync::Mutex;
+
+use crate::default_builder;
+
+use super::{CoreFetch, CoreFetchConfig, Transport};
+
+#[derive(Debug)]
+pub struct MockTransport {
+    requests_sent: Arc<Mutex<Vec<(OpId, AgentId)>>>,
```

Comment:
This contains simple types, should use std::sync::Mutex intead of the tokio one.

Reply:
Hidden in here https://github.com/holochain/kitsune2/pull/35/commits/05acb3110f09fffbe50ec38ad7d1ddc233e6f932